lib/Resizable.js

```diff
@@ -4,227 +4,492 @@ import PropTypes from 'prop-types';
+        }
+        if (max) {
+            width = Math.min(max[0], width);
+            height = Math.min(max[1], height);
+        }
+
+        // If the numbers changed, we must have introduced some slack. Record it for the next iteration.
+        slackW += (oldW - width);
+        slackH += (oldH - height);
+        if (slackW !== this.state.slackW || slackH !== this.state.slackH) {
+            this.setState({slackW, slackH});
+        }
+
+        return [width, height];
+    }
+
+    increaseHeightMoveUp = (currentTop, y, dimension, deltaY) => {
+        if (y < this.state.startY) {
+            let heightToAdd = Math.abs(deltaY)
+            if (this.props.maxConstraints[1] &&
+                heightToAdd + dimension.height > this.props.maxConstraints[1]) {
+                heightToAdd = this.props.maxConstraints[1] - dimension.height
+            }
+            dimension.height = this.state.height + heightToAdd
+            dimension.top = currentTop - heightToAdd
+        }
+        return dimension
+    }
+
+    reduceHeightMoveDown = (currentTop, y, dimension,deltaY) => {
+        if (y > this.state.startY) {
+
+            let heightToRemove = deltaY
+            if (this.props.minConstraints[1] &&
+                dimension.height - heightToRemove < this.props.minConstraints[1]) {
+                heightToRemove = 0
+            }
+            dimension.height = this.state.height - heightToRemove
+            dimension.top = currentTop + heightToRemove
+
+        }
+        return dimension
+    }
+
+    increaseWidthMoveLeft = (currentLeft, x, dimension,deltaX) => {
+        if (x < this.state.startX) {
+            let widthToAdd = Math.abs(deltaX)
+            if (this.props.maxConstraints[0] &&
+                widthToAdd + dimension.width > this.props.maxConstraints[0]) {
+                widthToAdd = this.props.maxConstraints[0] - dimension.width
+            }
+            dimension.width = this.state.width + widthToAdd
+            dimension.left = currentLeft - widthToAdd
+        }
+        return dimension
+    }
+
+    reduceWidthMoveRight = (currentLeft, x, dimension,deltaX) => {
+        if (x > this.state.startX) {
+            let widthToRemove = deltaX
+            if (this.props.minConstraints[1] &&
+                dimension.width - widthToRemove < this.props.minConstraints[1]) {
+                widthToRemove = 0
+            }
+            dimension.width = this.state.width - widthToRemove
+            dimension.left = currentLeft + widthToRemove
+        }
+        return dimension
+    }
+
+    resizeFromTopLeftCorner = (x, y, deltaX, deltaY) => {
+        const [currentLeft, currentTop] = this.getComponentPosition()
+        let dimension = this.getCurrentDimension()
+
+        dimension = (this.isMouseMovingUp(deltaY) ) ?
+            this.increaseHeightMoveUp(currentTop, y, dimension,deltaY) :
+            this.reduceHeightMoveDown(currentTop, y, dimension,deltaY)
```

Comment:
```
return this.isMouseMovingLeft(deltaX) ?
            this.increaseWidthMoveLeft(currentLeft, x, dimension,deltaX) :
            this.reduceWidthMoveRight(currentLeft, x, dimension,deltaX)
```
This block is repeated twice which means we can extract it (same for simular block in next function.

Reply:
done